src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -359,10 +360,18 @@ private boolean isChecksumValid(long value) {
         return (value & CRC32_CHECKSUM_SANITY) != 0;
     }
 
-    private VectorTransfer getVectorTransfer(VectorDataType vectorDataType) {
-        if (VectorDataType.BINARY == vectorDataType) {
-            return new VectorTransferByte(KNNSettings.getVectorStreamingMemoryLimit().getBytes());
+    private VectorTransfer getVectorTransfer(VectorDataType vectorDataType, KNNEngine knnEngine) {
+        long memoryLimit = KNNSettings.getVectorStreamingMemoryLimit().getBytes();
+        switch (vectorDataType) {
+            case BINARY:
+                return new VectorTransferBinary(memoryLimit);
+            case BYTE:
+                if (KNNEngine.FAISS == knnEngine) {
```

Comment:
I think this is only either faiss or nmslib. nmslib does not support byte. Therefore, we don't need this condition check.